scripts/publish.js

```diff
@@ -66,29 +81,18 @@ async function main() {
            * hasn't already been published
            */
           async (dirent) => {
-            if (!dirent.parent || !dirent.isDirectory()) {
-              return
-            }
-            const { name, version, private } = JSON.parse(
-              await fs.readFile(
-                path.join(
-                  ROOT,
-                  dirent.parent.name,
-                  dirent.name,
-                  'package.json'
-                ),
-                'utf8'
-              )
+            const { name, version, private } = await chillJSONParse(
+              fs.readFile(path.join(dirent.fullpath(), 'package.json'), 'utf8')
             )
-            if (private) {
+            if (private || !name) {
```

Comment:
if `name` is undefined, that's an invalid `package.json` file and we should probably say something